dev-tools/mage/integtest_docker.go

```diff
@@ -235,71 +451,34 @@ func integTestDockerComposeEnvVars() (map[string]string, error) {
 	}, nil
 }
 
-// dockerComposeProjectName returns the project name to use with docker-compose.
-// It is passed to docker-compose using the `-p` flag. And is passed to our
-// Go and Python testing libraries through the DOCKER_COMPOSE_PROJECT_NAME
-// environment variable.
-func dockerComposeProjectName() string {
-	commit, err := CommitHash()
-	if err != nil {
-		panic(errors.Wrap(err, "failed to construct docker compose project name"))
-	}
-
-	version, err := BeatQualifiedVersion()
-	if err != nil {
-		panic(errors.Wrap(err, "failed to construct docker compose project name"))
+// WriteDockerComposeEnvFile generates a docker-compose environment variable file.
+func WriteDockerComposeEnvFile() (string, error) {
+	envFileContent := []string{
+		"# Environment variable file to pass to docker-compose with the --env-file option.",
 	}
-	version = strings.NewReplacer(".", "_").Replace(version)
-
-	projectName := "{{.BeatName}}_{{.Version}}_{{.ShortCommit}}-{{.StackEnvironment}}"
-	projectName = MustExpand(projectName, map[string]interface{}{
-		"StackEnvironment": StackEnvironment,
-		"ShortCommit":      commit[:10],
-		"Version":          version,
-	})
-	return projectName
-}
-
-// dockerComposeBuildImages builds all images in the docker-compose.yml file.
-func dockerComposeBuildImages() error {
-	fmt.Println(">> Building docker images")
-
-	composeEnv, err := integTestDockerComposeEnvVars()
+	envVarMap, err := integTestDockerComposeEnvVars()
 	if err != nil {
-		return err
-	}
-
-	args := []string{"-p", dockerComposeProjectName(), "build", "--force-rm"}
-	if _, noCache := os.LookupEnv("DOCKER_NOCACHE"); noCache {
-		args = append(args, "--no-cache")
+		return "", err
 	}
 
-	if _, forcePull := os.LookupEnv("DOCKER_PULL"); forcePull {
-		args = append(args, "--pull")
+	for k, v := range envVarMap {
+		envFileContent = append(envFileContent, fmt.Sprintf("%s=%s", k, v))
 	}
 
-	out := ioutil.Discard
-	if mg.Verbose() {
-		out = os.Stderr
+	esBeatsDir, err := ElasticBeatsDir()
+	if err != nil {
+		return "", err
 	}
 
-	_, err = sh.Exec(
-		composeEnv,
-		out,
-		os.Stderr,
-		"docker-compose", args...,
+	envFile := path.Join(esBeatsDir, "docker.env")
+	err = os.WriteFile( //nolint:gosec // File permissions are not a security risk.
```

Comment:
In general, gosec seems to give pretty dubious advice.

Reply:
Yes, complaining about file permissions and suggesting you switch to the cryptographic random number generator are all I've seen it do. I am hesitant to turn it off though because of the small chance it finds a real issue. I haven't seen it do that yet though...

Reply:
What I would like to see in the general case (I don't think golangci-lint supports this), is to have two levels of linting, correctness linting and style linting differentiated on the possibility of false positives. gosec would fall into the latter case along with quite a few others that are currently turned on.

Reply:
@efd6 we could start maintaining 2 different configurations for golangci-lint to enable the 2 modes you described. However, that does not change the fact that the check would fail it just moves the failure to another place and you'll find out later than sooner. I personally don't see a point of doing this.

Reply:
My desire was to have the style-based linting not failout, but only annotate (for judgement by reviewers). I don't think this is possible. Currently, there are many cases where the linter is pushing authors to write worse code.

Reply:
@efd6 
> My desire was to have the style-based linting not failout, but only annotate (for judgement by reviewers). I don't think this is possible.

as I said, it is possible. We could run the linter twice (which would increase the runtime):

1. The configuration with style-based rules and the configuration option `issues-exit-code: 0` which will not fail the run (not sure that the current Github action would pick the issues up with exit 0 though)
2. The configuration with required checks that fails the check

However, mind you that the current linting check does not fail or interrupt the main CI pipeline, it works in parallel as a Github action that propagates the issues to the diff visualisation. And that's exactly the behaviour that you propose, is not it? Reviewers very often decide to ignore the linting checks and merge regardless the failed linting. AFAIK, linting is not a required check at the moment (if it is we can change that).

For the beats repository linting is a suggestion, we cannot address all the issues all the time because of the size of the codebase. For new repositories it should be required and reviewers should be more demanding there.

Reply:
UPD. Linting was a required check, I removed this requirement and now it does not block the merge anymore. Reviewers decide whether to request addressing the linting issues or not.

Reply:
> UPD. Linting was a required check, I removed this requirement and now it does not block the merge anymore. Reviewers decide whether to request addressing the linting issues or not.

I would rather see us reduce the number of linters we use than make it optional. Linters like `staticcheck` and `errcheck` have a very low false positive rate.

We'll move this conversation out of this review, likely I will open a PR on the linter set in beats later.